spec/features/invite_friend_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+
+feature "the invite a friend functionality" do 
+  scenario "the user invites a friend, the friend registers, the user and friend follow each other" do
+    inviter = Fabricate(:user)
+    sign_in(inviter) 
+
+    invite_friend_by_email
+
+    click_link 'Sign Out'
+
+    friend_clicks_link_in_email_invitation
+    friend_registers
+    friend_signs_in
+    friend_is_following_inviter(inviter)
+    
+    click_link 'Sign Out'
+
+    sign_in(inviter)
+    inviter_is_following_friend
+    
+    clear_email  
+  end
+
+  def invite_friend_by_email
+    click_link 'Invite a friend'
+    fill_in "Friend's Name", with: "Bob"
+    fill_in "Friend's Email Address", with: "bobmiller@example.com"
+    fill_in "Invitation Message", with: "I think you'll really like this site!"
+    click_button 'Send Invitation'
+  end 
+
+  def friend_clicks_link_in_email_invitation
+    open_email('bobmiller@example.com')
+    current_email.click_link('Join MyFlix')
+  end 
+
+  def friend_registers
+    fill_in 'Password', with: "bob"
+    fill_in 'Full Name', with: "Bob Miller"
+    click_button 'Sign Up'
+  end 
+
+  def friend_signs_in
+    fill_in 'Email Address', with: "bobmiller@example.com"
+    fill_in 'Password', with: 'bob'
+    click_button 'Sign in' 
+    expect(page).to have_content 'You have successfully signed in.' 
```

Comment:
Actually you don't have to validate whether used signed in right. I would cut the expectation - also the abstraction name does not say anything about validating sign in. My reason for this is - you are already testing signing in in a different feature spec. The expectation should be on a value critical for this workflow (as you have below).